CollisionOperator/CollisionProcessors/neutronCEimp_class.f90

```diff
@@ -246,13 +250,19 @@ subroutine sampleCollision(self, p, tally, collDat, thisCycle, nextCycle)
     if(.not.associated(self % mat)) call fatalError(Here, 'Material is not ceNeutronMaterial')
 
     ! Select collision nuclide
-    collDat % nucIdx = self % mat % sampleNuclide(p % E, p % pRNG)
+    call self % mat % sampleNuclide(p % E, p % pRNG, collDat % nucIdx, self % eSample)
```

Comment:
On eSample, is it more natural to have it in the collisionProcessor than the particle? I ask with respect to (someday!) not having to make thread private copies of the processor.

Reply:
Yes that's a fair point. Adding a property like eCollision to the particle would be also ok!

Reply:
But collDat is also a property of the processor right? So the issue about the need for a thread private processor would stay. 
I think it would be nicer to keep that energy within the collision processor, but the implementation with the particle should be safe for now because I expect that property to be only read and updated in the collision processor. I can't think of an ultimate solution here I think.

Reply:
Well if it isn't then I'm happy to revert back and save it in collDat!